### PR TITLE
A new data package has been added.

### PR DESCRIPTION
**Pull request by banchen19**
添加了一份新的数据包，参考于：

image

### CODE REVIEW

**Review by banchen19 — COMMENTED** [review]
o

File: crates/proto/src/types/actor_type.rs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        &self,
+        stream: &mut Vec<u8>,
+    ) -> Result<(), bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(())
+    }
+
+    fn proto_deserialize(
+        stream: &mut std::io::Cursor<&[u8]>,
+    ) -> Result<Self, bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(Self::Pig)
```
Comment (theaddonn) [inline crates/proto/src/types/actor_type.rs]: Are you sure that is correct?

File: crates/proto/src/types/actor_type.rs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    Bogged = Flags::BOGGED.bits(),
+    OminousItemSpawner = Flags::OMINOUSITEMSPAWNER.bits(),
+}
+
+impl ProtoCodec for ActorType {
+    fn proto_serialize(
+        &self,
+        stream: &mut Vec<u8>,
+    ) -> Result<(), bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(())
```
Comment (theaddonn) [inline crates/proto/src/types/actor_type.rs]: If you did not implement something, please just add `todo("implement this later)` or `unimplemented!()`. Rust has dedicated macros for this.

File: crates/proto/src/types/dataitem.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        &self,
+        stream: &mut Vec<u8>,
+    ) -> Result<(), bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(())
+    }
+
+    fn proto_deserialize(
+        stream: &mut std::io::Cursor<&[u8]>,
+    ) -> Result<Self, bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(Self::ValByte(0))
```
Comment (theaddonn) [inline crates/proto/src/types/dataitem.rs]: Same `unimplemented!()` stuff goes here

File: crates/proto/src/types/dataitem.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    ValInt64(LE<i64>),
+    ValVec3(Vec3<LE<f32>>),
+}
+use bedrockrs_proto_core::ProtoCodec;
+impl ProtoCodec for DataItemValue {
+    fn proto_serialize(
+        &self,
+        stream: &mut Vec<u8>,
+    ) -> Result<(), bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(())
```
Comment (theaddonn) [inline crates/proto/src/types/dataitem.rs]: Same `unimplemented!()` stuff goes here

File: crates/proto/src/types/property_sync_data.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    Float(FloatEntriesList),
+}
+
+use bedrockrs_proto_core::ProtoCodec;
+impl ProtoCodec for PropertySyncData {
+    fn proto_serialize(
+        &self,
+        stream: &mut Vec<u8>,
+    ) -> Result<(), bedrockrs_proto_core::error::ProtoCodecError> {
+        Ok(())
```
Comment (theaddonn) [inline crates/proto/src/types/property_sync_data.rs:39]: Same `unimplemented!()` stuff goes here

File: crates/proto/src/packets/add_actor_packet.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+pub struct AddActorPacket {
+    pub target_actor_id: ActorUniqueID,
+    pub target_runtime_id: ActorRuntimeID,
+    pub actor_type: ActorType,
+    pub position: Vec3<LE<f32>>,
+    pub velocity: Vec3<LE<f32>>,
+    pub rotation: Vec2<LE<f32>>,
+    pub y_head_rotation: LE<f32>,
+    pub y_body_rotation: LE<f32>,
+    pub attributes: AttributeList,
```
Comment (theaddonn) [inline crates/proto/src/packets/add_actor_packet.rs]: Why make a new struct, if you can directly do `Vec<Attribute>` (don't forget to add `#[len_repr(...)]`)

File: crates/proto/src/packets/add_actor_packet.rs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    pub actor_data: DataItemList,
```
Comment (theaddonn) [inline crates/proto/src/packets/add_actor_packet.rs]: Why make a new struct, if you can directly do `Vec<DataItem>` (don't forget to add `#[len_repr(...)]`)

File: crates/proto/src/packets/add_actor_packet.rs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    pub synched_properties: PropertySyncData,
+    pub actor_links: ActorLinkList,
```
Comment (theaddonn) [inline crates/proto/src/packets/add_actor_packet.rs]: Why make a new struct, if you can directly do `Vec<ActorLink>` (don't forget to add `#[len_repr(...)]`)

File: crates/proto/src/types/dataitem.rs [hunk 9]
```diff
@@ -0,0 +1,45 @@
+use bedrockrs_core::{
+    int::{LE, VAR},
+    Vec3,
+};
+use bedrockrs_nbt::NbtTag;
+use bedrockrs_proto_derive::ProtoCodec;
+
+#[derive(ProtoCodec, Debug, Clone)]
+struct DataItem {
+    id: VAR<u32>,
```
Comment (theaddonn) [inline crates/proto/src/types/dataitem.rs]: `pub` :3

File: crates/proto/src/types/dataitem.rs [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    value: DataItemValue,
```
Comment (theaddonn) [inline crates/proto/src/types/dataitem.rs]: `pub` :p

**Comment by banchen19** [comment]
Have all

**Pull request closed without merging**